python/vesin/tests/test_neighbors.py

```diff
@@ -16,20 +16,23 @@
 def test_neighbors(system, cutoff):
     atoms = ase.io.read(f"{CURRENT_DIR}/data/{system}.xyz")
 
-    ase_nl = ase.neighborlist.neighbor_list("ijSD", atoms, cutoff)
-    ase_nl = [(i, j, S, D) for i, j, S, D in zip(*ase_nl)]
-
-    vesin_nl = vesin.ase_neighbor_list("ijSD", atoms, cutoff)
-    vesin_nl = [(i, j, S, D) for i, j, S, D in zip(*vesin_nl)]
-
-    assert len(ase_nl) == len(vesin_nl)
-
-    for i, j, S, D in vesin_nl:
-        found = False
-        for ref_i, ref_j, ref_S, ref_D in ase_nl:
-            if i == ref_i and j == ref_j and np.all(S == ref_S):
-                assert np.allclose(D, ref_D)
-                found = True
-                break
-
-        assert found
+    ase_i, ase_j, ase_S, ase_D = ase.neighborlist.neighbor_list("ijSD", atoms, cutoff)
+    vesin_i, vesin_j, vesin_S, vesin_D = vesin.ase_neighbor_list("ijSD", atoms, cutoff)
+
+    assert len(ase_i) == len(vesin_i)
+    assert len(ase_j) == len(vesin_j)
+    assert len(ase_S) == len(vesin_S)
+    assert len(ase_D) == len(vesin_D)
+
+    ase_ijS = np.concatenate(
+        (ase_i.reshape(-1, 1), ase_j.reshape(-1, 1), ase_S), axis=1
+    )
+    vesin_ijS = np.concatenate(
+        (vesin_i.reshape(-1, 1), vesin_j.reshape(-1, 1), vesin_S), axis=1
+    )
+
+    ase_sort_indices = np.lexsort(np.rot90(ase_ijS))
```

Comment:
that's clever! Instead of `rot90`, can this be done by setting the right argument for `lexsort`'s `axis`? If not, that's fine =)

Reply:
No, `lexsort`'s `axis` seems to be doing something different than what we want (it doesn't work). I changed rot90 by a transpose for clarity